python/sdk/merlin/requirements.py

```diff
@@ -51,11 +51,16 @@ def process_conda_env(
         for additional_merlin_req in additional_merlin_reqs:
             exist = False
             for pip_req in pip_reqs:
-                pip_req_obj = Requirement(pip_req)
-                additional_merlin_req_obj = Requirement(additional_merlin_req)
-                if pip_req_obj.name.lower() == additional_merlin_req_obj.name.lower():
-                    exist = True
-                    break
+                # Skip requirement comparison for pip reqs that is not a package, which can happen if the user specifies
+                # repository or trusted host as part of the pip file
+                try:
+                    pip_req_obj = Requirement(pip_req)
+                    additional_merlin_req_obj = Requirement(additional_merlin_req)
+                    if pip_req_obj.name.lower() == additional_merlin_req_obj.name.lower():
+                        exist = True
+                        break
+                except InvalidRequirement:
```

Comment:
instead of doing a try-catch, would it be better to check and skip for entry starting with `--` (remove whitespace leading first)?

this will allow true invalid requirements to slip pass and defeat the purpose of any checking